### PR TITLE
Yr/create a new filter any match/ciac 6748

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-6748

Description
Adding a new filter.
This filter accepts two inputs, left and right, which can each be either a single value of all types (e.g., string, int, etc) or a list of values. The filter iterates over each value in the left input and checks if the entire value or part of it exists within the right input. The matching process is case-insensitive, meaning it disregards letter case during the comparison.

Moreover, the comparison treats all inputs as strings. Hence, integers and strings are considered equal during the evaluation.

## Must have
- [x] Tests
- [x] Documentation 

##Screenshots

AnyMatch_Test_Tue_Aug_01_2023

### CODE REVIEW

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch_test.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import pytest
+import demistomock as demisto
+from AnyMatch import main
+
+# Note: left and right can be all types, and in order to test list as an input we use a string that looks like a list,
+# a comma separated string that will be converted to a list in the script.
+
+
+@pytest.mark.parametrize('left,right, call_count,result', [
```
Comment (ShacharKidor) [inline Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch_test.py:9]:
```suggestion
@pytest.mark.parametrize('left,right, call_count,expected_result', [
```

**Review by ShacharKidor — CHANGES_REQUESTED** [review]
Good work!
Added several comments.

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+def main():
+    leftArg = demisto.args()["left"]
+    rightArg = demisto.args()["right"]
+
+    left_list = argToList(str(leftArg))
+    right_list = argToList(str(rightArg))
+
+    for left_val in left_list:
+        demisto.results(is_substring_in_list(left_val, right_list))
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.py:26]: Changed to return_results.
not sure abour the second note, this is the way it is in other filters, IsInCiderRange for exapmle.

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+def is_substring_in_list(single_str: str, str_list: list[str]) -> bool:
+    """
+    This function checks if a string is in a list of strings, fully or partly, case insensitive.
+    Args:
+       single_str: A string to check if it exists in the list.
+       str_list: A list of strings to check if the single_str is in it, fully or partly.
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.py:10]: Fixed.

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+def is_substring_in_list(single_str: str, str_list: list[str]) -> bool:
+    """
+    This function checks if a string is in a list of strings, fully or partly, case insensitive.
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.py:7]: Fixed.

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+args:
+  - name: left
+    required: true
+    isArray: true
+    description: Value to check if it exists in the right side, fully or partially. can be a single value or a comma-separated list.
+  - name: right
+    required: true
+    isArray: true
+    description: Value to check if it is equal to a string or substrings from the left. can be a single value or a comma-separated list.
+scripttarget: 0
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.yml:24]: Correct.No outputs in all examples I sow.

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+name: AnyMatch
+script: ""
+type: python
+subtype: python3
+tags:
+  - filter
+  - general
+comment: |-
+  Returns all elements from the left side that exists in the right side, fully or partially. Note: This filter is case insensitive.
+  e.g -AnyMatch left=[A,b,c] right=[baby] will return [a,b].
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.yml:13]: Fixed

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+enabled: true
+args:
+  - name: left
+    required: true
+    isArray: true
+    description: Value to check if it exists in the right side, fully or partially. can be a single value or a comma-separated list.
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.yml:19]: Fixed.

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch_test.py [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    (1, "1", 1, [True]),       # int and str are equal
+    ("bca", "A", 1, [True]),  # case insensitive
+    ("ABC", "a", 1, [True]),  # case insensitive
+    ({"alert": {"data": "x"}}, "x", 1, [True]),
+    ("{'a':1,'c':2}", "{'a': 1}, {'b': 2}", 2, [False, False]),     # {'a':1} is not a part of {'a':1, or 'c':2}
+    ("{'a': 1}, {'b': 2}", "{a:1}", 2, [False, False]),  # {a:1} is not a part of {'a': 1} or {'b': 2}
+    # although '' is not a part of {'a':1,'c':2}, but ' is in {'a': 1 and in  'c': 2}
+    ("{'a':1,'c':2}", "'', '", 2, [True, True]),
+])
+def test_main(mocker, left, right, call_count, result):
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch_test.py:25]: Fixed

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/README.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
+This filter accepts two inputs, left and right, each of which can be either a single value of any type (e.g., string, int, etc.) or a list of values. The filter iterates over each value in the left input and returns all values that exist within the right input, even if only a part of it exists. The matching process is case-insensitive, meaning it disregards letter case during the comparison.
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/FiltersAndTransformers/Scripts/AnyMatch/README.md:1]: Fixed.

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/README.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+| Tags                 | filter          |
+| Cortex XSOAR Version | 6.9.0           |
+
+## Inputs
+
+---
+
+| **Argument Name** | **Description**                                                                                                    |
+| ----------------- | ------------------------------------------------------------------------------------------------------------------ |
+| left              | Value to check if it exists in the right side, fully or partially. can be a single value or a comma-separated list.                    |
```
[the remark this answers is not in the document]
Comment (RosenbergYehuda) [reply Packs/FiltersAndTransformers/Scripts/AnyMatch/README.md:25]: Fixed.

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ShacharKidor) [reply Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.py]: ok in that case, let's stick to that.

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.yml [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+name: AnyMatch
+script: ""
+type: python
+subtype: python3
+tags:
+  - filter
+  - general
+comment: |-
+  Returns all elements from the left side that have a substring that is equal to an element from the right side. Note: This filter is case-insensitive.
+  e.g -AnyMatch left=baby right=A will return baby. For more examples see the filter's Readme.
```
Comment (ShirleyDenkberg) [inline Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.yml:13]:
```suggestion
  E.g., -AnyMatch left=baby right=I will return baby. For more examples see the filter's Readme.
```

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.yml [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+enabled: true
+args:
+  - name: left
+    required: true
+    isArray: true
+    description: Value to check if it has a substring that is equal to an element in the right side. can be a single value or a comma-separated list.
```
Comment (ShirleyDenkberg) [inline Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.yml:19]:
```suggestion
    description: Value to check if it has a substring that is equal to an element in the right side. Can be a single value or a comma-separated list.
```

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.yml [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+  - name: right
+    required: true
+    isArray: true
+    description: Value to check if it is equal to an element or to a substring of an element from the left. can be a single value or a comma-separated list.
```
Comment (ShirleyDenkberg) [inline Packs/FiltersAndTransformers/Scripts/AnyMatch/AnyMatch.yml:23]:
```suggestion
    description: Value to check if it is equal to an element or to a substring of an element from the left. Can be a single value or a comma-separated list.
```

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/README.md [hunk 14]
```diff
@@ -0,0 +1,49 @@
+This filter accepts two inputs, left and right, each of which can be either a single element of any type (e.g., string, int, etc.) or a list of elements. The filter iterates over each element in the left input and returns all elements that have a substring that is equal to an element from the right side. The matching process is case-insensitive, meaning it disregards letter case during the comparison.
+
+All inputs are treated either as a string or as a list of strings, if it contains a comma. Json on the other hand is always treated as one string.
```
Comment (ShirleyDenkberg) [inline Packs/FiltersAndTransformers/Scripts/AnyMatch/README.md:3]:
```suggestion
All inputs are treated either as a string or as a list of strings, if it contains a comma. A JSON is always treated as a string.
```

File: Packs/FiltersAndTransformers/Scripts/AnyMatch/README.md [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+| Tags                 | filter          |
+| Cortex XSOAR Version | 6.9.0           |
+
+## Inputs
+
+---
+
+| **Argument Name** | **Description**                                                                                                    |
+| ----------------- | ------------------------------------------------------------------------------------------------------------------ |
+| left              | Value to check if it has a substring that is equal to an element in the right side. can be a single value or a comma-separated list.                   |
```
Comment (ShirleyDenkberg) [inline Packs/FiltersAndTransformers/Scripts/AnyMatch/README.md:25]:
```suggestion
| left              | Value to check if it has a substring that is equal to an element in the right side. Can be a single value or a comma-separated list.                   |
```

**Comment by ShirleyDenkberg** [comment]
@ShacharKidor @ostolero Doc review completed.

**Pull request merged**